src/coreclr/jit/morph.cpp

```diff
@@ -13822,7 +13900,8 @@ PhaseStatus Compiler::fgMorphBlocks()
 
     // Local assertion prop is enabled if we are optimized
     //
-    optLocalAssertionProp = opts.OptimizationEnabled();
+    optLocalAssertionProp           = opts.OptimizationEnabled();
+    optCrossBlockLocalAssertionProp = optLocalAssertionProp;
```

Comment:
```suggestion
    optCrossBlockLocalAssertionProp = optLocalAssertionProp && JitConfig.JitDoCrossBlockLocalAssertionProp();
```
?

Reply:
I see you consult the config in `optAssertionInit`. Is there a reason to wait until then?

Reply:
I went back and forth as to whether all this setup should be assertion prop's business; I could push more of the logic there I suppose.